pkg/agent/openflow/network_policy.go

```diff
@@ -1775,33 +1816,97 @@ func parseMetricFlow(flow string) (uint32, types.RuleMetric) {
 	return parseAllowFlow(flowMap)
 }
 
-func (c *client) NetworkPolicyMetrics() map[uint32]*types.RuleMetric {
+func (c *client) MulticastIngressPodMetrics() map[uint32]*types.RuleMetric {
 	result := map[uint32]*types.RuleMetric{}
-	egressFlows, _ := c.ovsctlClient.DumpTableFlows(EgressMetricTable.ofTable.GetID())
-	ingressFlows, _ := c.ovsctlClient.DumpTableFlows(IngressMetricTable.ofTable.GetID())
+	ingressFlows, _ := c.ovsctlClient.DumpTableFlows(MulticastIngressPodMetricTable.ofTable.GetID())
+	for _, flow := range ingressFlows {
+		if !strings.Contains(flow, metricFlowIdentifier) {
+			continue
+		}
+		flowMap := parseFlowToMap(flow)
+		ruleID, metric := parseMulticastIngressPodFlow(flowMap)
+		if accMetric, ok := result[ruleID]; ok {
+			accMetric.Merge(&metric)
+		} else {
+			result[ruleID] = &metric
+		}
+	}
+	return result
+}
+
+func (c *client) MulticastEgressPodMetricsBySrc(ip net.IP) *types.RuleMetric {
+	table := MulticastEgressPodMetricTable.ofTable.GetID()
+	nw_src := ip.String()
+	flow, _ := c.ovsctlClient.DumpMatchedFlow(fmt.Sprintf("table=%d,ip,nw_src=%s,nw_dst=224.0.0.0/4", table, nw_src))
+	if len(flow) == 0 {
+		return &types.RuleMetric{}
+	}
+	flowMap := parseFlowToMap(flow)
+	metric := parseFlowMetric(flowMap)
+	return &metric
+}
+
+func (c *client) MulticastIngressPodMetricsByReg1(ofPort int32) *types.RuleMetric {
+	table := MulticastIngressPodMetricTable.ofTable.GetID()
+	reg1 := ofPort
+	flow, _ := c.ovsctlClient.DumpMatchedFlow(fmt.Sprintf("table=%d,ip,reg1=%d", table, reg1))
```

Comment:
Since IGMP query metric flow is also installed in this table, this match condition has no "strict" flag, it should match all flows targeting at the Pod openflow port, is it what you really want?

Reply:
IGMP query metric flow is installed in MulticastIngressMetric and it has no reg field. It won't be matched here